apps/communications/contacts/js/activities_vcard.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+/* global LazyLoader, VCardReader, Contacts */
+/* exported VcardActivityHandler */
+
+var VcardActivityHandler = function() {
+  var handle = (activity, parentHandler) => {
+    const DEPENDENCIES = '/shared/js/contacts/import/utilities/vcard_reader.js';
+    LazyLoader.load(DEPENDENCIES, function () {
+      getvCardReader(activity.source.data.blob).then(readvCard, (err) => {
+        console.error(err);
+        Contacts.cancel();
+      });
+    });
+
+    function getvCardReader(blob) {
+      return new Promise((resolve, reject) => {
+        var fileReader = new FileReader();
+        fileReader.readAsText(blob);
+        fileReader.onloadend = () => {
+          resolve(new VCardReader(fileReader.result));
+        };
+        fileReader.onerror = () => {
+          console.error('The blob could not be read as a file.');
+          reject(fileReader.error);
+        };
+      });
+    }
+
+    function readvCard(vCardReader) {
+      var firstContact;
+      var cursor = vCardReader.getAll();
+      cursor.onsuccess = (event) => {
+        var contact = event.target.result;
+        // We check if there is only one contact to know what function
+        // we should call. If not, we render the contacts one by one.
+        if (contact) {
+          if (!firstContact) {
+            firstContact = contact;
+            cursor.continue();
+          } else {
+            render([firstContact, contact], activity, cursor);
+          }
+        } else if (firstContact) {
+          renderOneContact(firstContact, activity);
+        }
+      };
+
+      cursor.onerror = () => {
+        console.error('The vCard could not be read');
```

Comment:
In fact, I have been looking in the vCardReader library and an onerror handler is not defined, so I will remove that handler.